tests/integration/tests.sh

```diff
@@ -6,55 +6,45 @@
 # TODO: Rewrite in Python.
 #
 
-set -e
+set -euo pipefail
 
 # Make filenames specific to Python version. This way the output of tox tests is separated.
 python_version=$(python --version | tail -c +8)
-actual_filename="actual-$python_version.yml"
-errors_filename="errors-$python_version.log"
-diff_filename="diff-$python_version.log"
+actual_filename="actual-${python_version}.yml"
+errors_filename="errors-${python_version}.log"
 
 passed=0
 
-function trim() {
-  xargs echo -n
-}
-
 for test_scenario in tests/integration/examples/*
 do
-  cd "$test_scenario"
-
-  set +e
-  gadk --print 1> $actual_filename 2> $errors_filename
-  set -e
+  cd "${test_scenario}"
 
   # Confirm that there are no errors.
-  if [[ $(wc -l $errors_filename | trim) == "0 $errors_filename" ]]
+  if gadk --print > "${actual_filename}"
   then
-    echo "No errors in $test_scenario"
+    echo "No errors in ${test_scenario}"
   else
-    echo "Errors printed for $test_scenario"
-    cat $errors_filename
+    echo "Errors printed for ${test_scenario}"
+    cat "${errors_filename}"
     passed=1
     cd - > /dev/null
     continue
   fi
 
   # Confirm that actual.yml is not empty.
-  if [[ $(wc -l $actual_filename | trim) == "0 $actual_filename" ]]
+  if ! [[ -s "${actual_filename}" ]]
   then
-    echo "$test_scenario seems to be WIP!"
+    echo "${test_scenario} seems to be WIP!"
     passed=1
     cd - > /dev/null
     continue
   else
     # Confirm that actual.yml matches expected.yml.
-    diff expected.yml $actual_filename > $diff_filename
-    if [[ $(wc -l $diff_filename | trim) == "0 $diff_filename" ]]
+    if diff -U3 expected.yml "${actual_filename}"
```

Comment:
According to `man diff`, `-U3` is the default. Are you just being verbose here, or do you expect it to change based on `diff` version or something?

Reply:
[`man diff`][1] doesn't mention `-U3` - or do you mean that `3` is the default value for `-U`? In that case yes, I'm just being explicit. (The default is a "normal diff" which I find much harder to read.)

[1]: https://man7.org/linux/man-pages/man1/diff.1.html